src/sentry/notifications/notificationcontroller.py

```diff
@@ -169,224 +171,223 @@ def _get_layered_setting_options(
             setting_type: If specified, only return settings of this type.
         """
-        settings = self.get_all_setting_options
-        project_settings = self._filter_options(
-            settings=settings, scope_type=NotificationScopeEnum.PROJECT.value, **kwargs
-        )
-        org_settings = self._filter_options(
-            settings=settings, scope_type=NotificationScopeEnum.ORGANIZATION.value, **kwargs
-        )
-        user_settings = self._filter_options(
-            settings=settings, scope_type=NotificationScopeEnum.USER.value, **kwargs
+
+        if self.project_ids and len(list(self.project_ids)) > 2 and not project_id:
+            raise Exception("Must specify project_id if controller has more than 2 projects")
+
+        most_specific_setting_options: MutableMapping[
+            Recipient, MutableMapping[NotificationSettingEnum, NotificationSettingsOptionEnum]
+        ] = defaultdict(
+            lambda: defaultdict(
+                lambda: NotificationSettingsOptionEnum.DEFAULT
+            )  # Use lambda to return the default enum value
         )
-        scoped_settings = [project_settings, org_settings, user_settings]
 
-        defaults = get_type_defaults()
-        layered_setting_options: MutableMapping[
-            Recipient,
-            MutableMapping[
-                Scope, MutableMapping[NotificationSettingEnum, NotificationSettingsOptionEnum]
-            ],
-        ] = defaultdict(lambda: defaultdict(dict))
         for recipient in self.recipients:
-            (scope_type, scope_items) = self._get_scope_for_layering(recipient)
-            for item in scope_items:
-                scope: Scope = (scope_type, item)
-                for type in NotificationSettingEnum:
-                    most_specific_setting = None
-                    for setting in scoped_settings:
-                        user_or_team_settings = []
-                        if recipient_is_user(recipient):
-                            user_or_team_settings = self._filter_options(
-                                settings=setting, user_id=recipient.id, type=type.value
-                            )
-                        elif recipient_is_team(recipient):
-                            user_or_team_settings = self._filter_options(
-                                settings=setting, team_id=recipient.id, type=type.value
-                            )
-
-                        if len(user_or_team_settings) > 0:
-                            most_specific_setting = NotificationSettingsOptionEnum(
-                                user_or_team_settings[0].value
-                            )
-                            break
-
-                    if most_specific_setting is None and type in defaults:
-                        most_specific_setting = defaults[type]
-
-                    if most_specific_setting is None:
+            # get the settings for this user/team
+            filter_kwargs = kwargs.copy()
```

Comment:
Why do we need to copy the kwargs here?

Reply:
@snigdhas need a different copy of kwargs per user that is being queried